akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -210,6 +211,179 @@ final class PrefixAndTail[T](n: Int) extends GraphStage[FlowShape[T, (immutable.
 /**
  * INTERNAL API
  */
+final class GroupBy[T, K](maxSubstreams: Int, keyFor: T ⇒ K) extends GraphStage[FlowShape[T, Source[T, NotUsed]]] {
+  val in: Inlet[T] = Inlet("GroupBy.in")
+  val out: Outlet[Source[T, NotUsed]] = Outlet("GroupBy.out")
+  override val shape: FlowShape[T, Source[T, NotUsed]] = FlowShape(in, out)
+  override def initialAttributes = DefaultAttributes.groupBy
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new TimerGraphStageLogic(shape) with OutHandler with InHandler {
+    parent ⇒
+    lazy val decider = inheritedAttributes.get[SupervisionStrategy].map(_.decider).getOrElse(Supervision.stoppingDecider)
+    private val activeSubstreamsMap = new java.util.HashMap[Any, SubstreamSource]()
+    private val closedSubstreams = new java.util.HashSet[Any]()
+    private var timeout: FiniteDuration = _
+    private var substreamWaitingToBePushed: Option[SubstreamSource] = None
+    private var nextElementKey: K = null.asInstanceOf[K]
+    private var nextElementValue: T = null.asInstanceOf[T]
+    private var _nextId = 0
+    private val substreamsJustStared = new java.util.HashSet[Any]()
+    private var firstPushCounter: Int = 0
+
+    private def nextId(): Long = { _nextId += 1; _nextId }
+
+    private def hasNextElement = nextElementKey != null
+
+    private def clearNextElement(): Unit = {
+      nextElementKey = null.asInstanceOf[K]
+      nextElementValue = null.asInstanceOf[T]
+    }
+
+    private def tryCompleteAll(): Boolean =
+      if (activeSubstreamsMap.isEmpty || (!hasNextElement && firstPushCounter == 0)) {
+        for (value ← activeSubstreamsMap.values()) value.complete()
```

Comment:
Didn't know that. Thought it was always `map`/`flatMap` and just learnt about Alt+Enter in IntelliJ. :)

Gues it's because of the missing `yield`. Thx @johanandren